diff --git a/core/lib/Drupal/Core/Datetime/TimeZoneFormHelper.php b/core/lib/Drupal/Core/Datetime/TimeZoneFormHelper.php
index e79d6cbe1a..a6e6eedbda 100644
--- a/core/lib/Drupal/Core/Datetime/TimeZoneFormHelper.php
+++ b/core/lib/Drupal/Core/Datetime/TimeZoneFormHelper.php
@@ -69,6 +69,11 @@ public static function getOptionsListByRegion(bool $blank = FALSE): array {
         asort($grouped_zones[$key]);
       }
     }
+
+    // Allow modules to alter this list. But can't modify when submitting.
+    if (!isset($_POST['op'])) {
+      $grouped_zones = \Drupal::moduleHandler()->invokeAll('timezone_list_alter', [$grouped_zones]);
+    }
     return $grouped_zones;
   }
 
