diff --git a/core/modules/system/system.module b/core/modules/system/system.module
index 158068252e..71fd5a57d4 100644
--- a/core/modules/system/system.module
+++ b/core/modules/system/system.module
@@ -1138,6 +1138,9 @@ function system_time_zones($blank = NULL, $grouped = FALSE) {
     $zones = $grouped_zones;
   }
 
+  // Allow modules to alter this list.
+  $zones = \Drupal::moduleHandler()->invokeAll('timezone_list_alter', [$zones]);
+
   return $zones;
 }
 
